### PR TITLE
[Sival, spi dev] Flash mode test

### PR DESCRIPTION
**Pull request by engdoreis**
Link `chip_sw_spi_device_flash_mode` to `//sw/device/silicon_creator/rom/e2e:rom_e2e_smoke`.

### CODE REVIEW

File: sw/device/silicon_creator/rom/e2e/BUILD [hunk 1]
```diff
@@ -1,7 +1,10 @@
 # Copyright lowRISC contributors.
 # Licensed under the Apache License, Version 2.0, see LICENSE for details.
 # SPDX-License-Identifier: Apache-2.0
-
```
Comment (engdoreis) [inline sw/device/silicon_creator/rom/e2e/BUILD:4]: I'll revert this before merging, I don't want to trigger CI before getting reviews.

**Review by pamaury — APPROVED** [review]

File: hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson [hunk 2]
```diff
@@ -26,7 +26,7 @@
         "SPI_DEVICE.HW.LAST_READ_ADDR",
```
Comment (a-will) [inline hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson:26]: I'll just point out that the listed tests don't cover the full feature set. There are no reads during bootstrapping.

**Review by a-will — APPROVED** [review]

File: hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson (hunk 2 above)
Comment (engdoreis) [reply hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson]: Thanks for pointing this out. I split the test into two test, the new test will be a spi dev flash smoke test.

Comment (a-will) [reply hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson]: We actually don't have any chip-level tests that exercise this feature. It's also not active in Passthrough mode (only Flash mode, which doesn't do passthrough).

Comment (engdoreis) [reply hw/top_earlgrey/data/ip/chip_spi_device_testplan.hjson]: I noticed that when I started looking at the passthrought test, I didn't close the [issue](https://github.com/lowRISC/opentitan/issues/20631), so I'll create a spi_device_smoketest to cover that.

**Pull request merged**